server/resp/src/main/java/org/infinispan/server/resp/serialization/lua/LuaResponseWriter.java

```diff
@@ -0,0 +1,221 @@
+package org.infinispan.server.resp.serialization.lua;
+
+import static org.infinispan.server.resp.scripting.LuaContext.filterCause;
+import static org.infinispan.server.resp.scripting.LuaContext.luaPushError;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import org.infinispan.commons.jdkspecific.CallerId;
+import org.infinispan.server.resp.serialization.JavaObjectSerializer;
+import org.infinispan.server.resp.serialization.Resp3Type;
+import org.infinispan.server.resp.serialization.ResponseWriter;
+import org.infinispan.server.resp.serialization.SerializationHint;
+
+import party.iroiro.luajava.Lua;
+
+public class LuaResponseWriter implements ResponseWriter {
+   private final Lua lua;
+
+   public LuaResponseWriter(Lua lua) {
+      this.lua = lua;
+   }
+
+   @Override
+   public boolean isInternal() {
+      return true;
+   }
+
+   @Override
+   public void nulls() {
+      lua.checkStack(1);
+      lua.pushNil();
+   }
+
+   @Override
+   public void ok() {
+      lua.checkStack(1);
+      lua.push("OK");
+   }
+
+   @Override
+   public void queued(Object ignore) {
+      throw new UnsupportedOperationException("queued(Object)");
+   }
+
+   @Override
+   public void simpleString(CharSequence value) {
```

Comment:
I guess these string, integer, and double methods needs to check for null.

Reply:
I've added null checks to all methods